fem/eltrans.hpp

```diff
@@ -373,6 +382,199 @@ class InverseElementTransformation
    virtual int Transform(const Vector &pt, IntegrationPoint &ip);
 };
 
+/**
+ * @brief Performs batch inverse element transforms. Currently only supports
+ * meshes non-mixed meshes with SEGMENT, SQUARE, or CUBE geometries. Mixed
+ * element order meshes are projected onto an equivalent uniform order mesh.
+ */
+class BatchInverseElementTransformation
+{
+   // nodes grid function, not owned
+   const GridFunction *gf_ = nullptr;
+   // initial guess algorithm to use
+   InverseElementTransformation::InitGuessType init_guess_type =
+      InverseElementTransformation::ClosestPhysNode;
+   // num_1D_qpts = max(trans_order+rel_qpts_order,0)+1
+   int rel_qpts_order = 0;
+   // solution strategy to use
+   InverseElementTransformation::SolverType solver_type =
+      InverseElementTransformation::NewtonElementProject;
+   // basis type stored in points1d
+   int basis_type = BasisType::Invalid;
+   // initial guess points type. Quadrature1D::Invalid is used for match
+   // basis_type.
+   int guess_points_type = Quadrature1D::Invalid;
+   // max. number of Newton iterations
+   int max_iter = 16;
+   // internal element node locations cache
+   Vector node_pos;
+#ifdef MFEM_USE_DOUBLE
+   // reference space tolerance
+   real_t ref_tol = 1e-15;
+   // physical space tolerance (relative)
+   real_t phys_rtol = 4e-15;
+#else
+   // reference space tolerance
+   real_t ref_tol = 4e-7;
+   // physical space tolerance (relative)
+   real_t phys_rtol = 1e-6;
+#endif
+   // not owned, location of tensor product basis nodes in reference space
+   const Array<real_t> *points1d = nullptr;
+
+public:
+   BatchInverseElementTransformation();
+   BatchInverseElementTransformation(const GridFunction &gf,
```

Comment:
Document that the expected gridfunction is nodes. I would even use `nodes` instead of `gf`. I would still also add a constructor that takes the mesh as it is more intuitive.